tests/test_odcs_requester.py

```diff
@@ -27,19 +28,28 @@ def create_odcs_mock(self, compose_url: str) -> Callable[[int, bool], MagicMock]
         """Create an ODCS mock with specific results for the compose method"""
 
         def _mock_odcs_compose(
-            num_of_composes: int = 1, exception: bool = False
+            num_of_composes: int = 1, exception: bool = False, failed: bool = False
```

Comment:
Not sure I want to add *more* conditionals to it, but we should note this does not let us test a case where some composes passed and some failed.  
maybe we can change `failed` to `failed_composes`, and if it isn't `bool(failed_composes)` - generate as many failed composes as needed, same as we are doing with the successful ones.

Reply:
Changed. See if you like it better.